src/main/webapp/app/entities/genomic-indicator/genomic-indicator.tsx

```diff
@@ -7,19 +7,60 @@ import { ENTITY_ACTION, ENTITY_TYPE } from 'app/config/constants/constants';
 
 import { IRootStore } from 'app/stores';
 import EntityActionButton from 'app/shared/button/EntityActionButton';
-import { getEntityTableActionsColumn } from 'app/shared/util/utils';
-import OncoKBTable from 'app/shared/table/OncoKBTable';
+import { filterByKeyword, getAlterationName, getEntityTableActionsColumn } from 'app/shared/util/utils';
+import OncoKBTable, { SearchColumn } from 'app/shared/table/OncoKBTable';
 import { Column } from 'react-table';
+import { Association } from 'app/shared/api/generated';
+import { ISeqRegion } from 'app/shared/model/seq-region.model';
+import { IAssociation } from 'app/shared/model/association.model';
+import { IAlleleState } from 'app/shared/model/allele-state.model';
 
 export interface IGenomicIndicatorProps extends StoreProps, RouteComponentProps<{ url: string }> {}
 
+const getAssociatedAlterations = (associations?: IAssociation[]) => {
+  return associations
+    ?.map(association =>
+      association.alterations?.map(alt => `${alt.genes?.map(gene => gene.hugoSymbol).join('-')} ${alt.alteration}`).join()
+    )
+    .join();
+};
+
+const getAssociatedAlleleStates = (alleleStates?: IAlleleState[]) => {
+  return alleleStates?.map(alleleState => alleleState.name).join();
+};
+
 export const GenomicIndicator = (props: IGenomicIndicatorProps) => {
   useEffect(() => {
     props.getEntities({});
   }, []);
-  const columns: Column<IGenomicIndicator>[] = [
+  const columns: SearchColumn<IGenomicIndicator>[] = [
     { accessor: 'type', Header: 'Type' },
-    { accessor: 'name', Header: 'Name' },
+    {
+      accessor: 'uuid',
+      Header: 'UUID',
+      onFilter: (data: IGenomicIndicator, keyword) => (data.uuid ? filterByKeyword(data.uuid, keyword) : false),
+    },
+    {
+      accessor: 'name',
+      Header: 'Name',
+      onFilter: (data: IGenomicIndicator, keyword) => (data.name ? filterByKeyword(data.name, keyword) : false),
+    },
+    {
+      Header: 'Associated Allele States',
+      Cell(cell: { original }): JSX.Element {
+        return <span>{getAssociatedAlleleStates(cell.original.alleleStates)}</span>;
+      },
+      onFilter: (data: IGenomicIndicator, keyword) =>
+        data.alleleStates ? filterByKeyword(getAssociatedAlleleStates(data.alleleStates), keyword) : false,
+    },
+    {
+      Header: 'Associated Alterations',
```

Comment:
Should add `id` when we don't have `accessor`.

Reply:
oh, ok. Will do. Is it required?

Reply:
Only required if accessor is not defined. I believe the sorting will also not work for `Associated Allele States` and `Associated Alterations` because the sorting by default will try to use the value that is found using the accessor.

Reply:
Added the accessor so it can be sorted.